test_project/fastpg/db.py

```
@@ -84,34 +84,58 @@ async def fetch_all(self, query: str, values: Optional[Dict] = None):
+            await transaction_ctx.rollback()
             raise e
         else:
-            await transaction.commit()
+            await transaction_ctx.commit()
         return result
 
+    async def transaction(self):
+        """Return a database transaction object managed by the caller."""
+
+        if not self.database:
+            raise RuntimeError("Database connection is not initialized")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)